### ISSUE TITLE
About training EDSR on multi GPUs

### ISSUE THREAD

**Issue by UESTCshiny** [open-mmlab/mmsr#37]
Hello, the official training configuration of EDVR is on 8 GPUs. If I train the model on only 2 1080Ti, what should I set the learning rate to?  In other words, what is the relationship between lr and the number of GPUs?

**Comment by BaohaoLiao** [comment]
> Hello, the official training configuration of EDVR is on 8 GPUs. If I train the model on only 2 1080Ti, what should I set the learning rate to? In other words, what is the relationship between lr and the number of GPUs?

I think the best way to reproduce results with less GPUs is to accumulate the gradients, i.e. keep the the total batch size of update the same (num_GPU * batch_size stays the same). I haven't find these codes in mmsr. So you need to write them by yourself.

**Comment by m732367606** [comment]
If you want to train a model with 2 gpu, just modify the train.yml file.
Modify the gpu_ids: [0] to gpu_ids: [0,1] is OK.

**Comment by ascetic-yuanma** [comment]
If you want to train your model with more than 1 GPU, you can modify the gpu_ids parameter.
Such as follows,
gpu_ids: [0] ---> gpu_ids: [0,1]
for 2 GPUs.
Hope this can help you!

**Comment by TomArrow** [comment]
What kind of settings would I need for single GPU training with a GTX 1070 with 6GB VRAM?

**Comment by ascetic-yuanma** [comment]
> What kind of settings would I need for single GPU training with a GTX 1070 with 6GB VRAM?
> 
> Edit: I mean 8 GB VRAM

Set gpu_ids: [0] is OK.
Hope this can help you!

**Comment by vegetable09** [comment]
This work for me:CUDA_VISIBLE_DEVICES=0\,1 python -m torch.distributed.launch  --nproc_per_node=2 --master_port=14321 /workdir/mmsr/codes/train.py -opt /workdir/mmsr/codes/options/train/train_ESRGAN_parallel.yml --launcher pytorch
